ynr/apps/bulk_adding/templates/bulk_add/sopns/add_review_form.html

```diff
@@ -74,7 +77,15 @@ <h4><em>“{{ form.party_description_text.value }}”</em></h4>
                   {% for candidacy in choice.choice_label.previous_candidacies %}
                     <li>{{ candidacy }}</li>
                   {% endfor %}
-
+                </td>
+                <td>
+                {% if ballot.is_welsh_run %} 
+                  {% for candidacy in choice.choice_label.previous_candidacies %}
+                    {% for party in candidacy.previous_party_affiliations.all %}
+                      <li>{{ party }}</li>
```

Comment:
What am I missing here?

Reply:
I think the TL;DR is that I dont think we need this here and we should remove it.

Hmm this is an odd one, I ran it locally to have a play around, and actually I dont think we want/need to display this here as I dont think it is useful information. As this is the form that we display to users to help them avoid creating duplicate candidates so:
1. It might actually be more confusing than useful
2. Its rare there would be anything to display anyway, and for this election won't be anything as this is all new data we are capturing for the first time for these elections.

Reply:
Where we do want to display this is on the "Existing people" table above this, which is the `known_people` template. I think the easiest way to do this is to copy how the party is assigned to the `person` [object in the view](https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/bulk_adding/views/sopns.py#L74)

So if we added this:
```
person.previous_party_affiliations = membership.previous_party_affiliations.all()
```
Then you can then loop through them in the template with:

```
            {% if person.previous_party_affiliations %}
              <td>
                  {% for party in person.previous_party_affiliations.all %}
                    {{ party.name }}
                  {% endfor %}
              </td>
            {% endif %}
```